b2sdk/transfer/emerge/executor.py

```diff
@@ -0,0 +1,480 @@
+        bucket_id,
+        file_name,
+        content_type,
+        file_info,
+        progress_listener,
+        continue_large_file_id=None,
+        max_queue_size=None,
+    ):
+        if emerge_plan.is_large_file():
+            execution = LargeFileEmergeExecution(
+                self.session,
+                self.services,
+                bucket_id,
+                file_name,
+                content_type,
+                file_info,
+                progress_listener,
+                continue_large_file_id=continue_large_file_id,
+                max_queue_size=max_queue_size,
+            )
+        else:
+            if continue_large_file_id is not None:
+                raise ValueError('Cannot resume emerging single part plan.')
+            execution = SmallFileEmergeExecution(
+                self.session,
+                self.services,
+                bucket_id,
+                file_name,
+                content_type,
+                file_info,
+                progress_listener,
+            )
+        return execution.execute_plan(emerge_plan)
+
+
+@six.add_metaclass(ABCMeta)
+class BaseEmergeExecution(object):
+    DEFAULT_CONTENT_TYPE = AUTO_CONTENT_TYPE
+
+    def __init__(
+        self, session, services, bucket_id, file_name, content_type, file_info, progress_listener
+    ):
+        self.session = session
+        self.services = services
+        self.bucket_id = bucket_id
+        self.file_name = file_name
+        self.content_type = content_type
+        self.file_info = file_info
+        self.progress_listener = progress_listener
+
+    @abstractmethod
+    def execute_plan(self, emerge_plan):
+        pass
+
+
+class SmallFileEmergeExecution(BaseEmergeExecution):
+    def execute_plan(self, emerge_plan):
+        emerge_parts = list(emerge_plan.emerge_parts)
+        assert len(emerge_parts) == 1
+        emerge_part = emerge_parts[0]
+        execution_step_factory = SmallFileEmergeExecutionStepFactory(self, emerge_part)
+        execution_step = execution_step_factory.get_execution_step()
+        future = execution_step.execute()
+        return future.result()
+
+
+class LargeFileEmergeExecution(BaseEmergeExecution):
+    MAX_LARGE_FILE_SIZE = 10 * 1000 * 1000 * 1000 * 1000  # 10 TB
+
+    def __init__(
+        self,
+        session,
+        services,
+        bucket_id,
+        file_name,
+        content_type,
+        file_info,
+        progress_listener,
+        continue_large_file_id=None,
+        max_queue_size=None,
+    ):
+        super(LargeFileEmergeExecution, self).__init__(
+            session, services, bucket_id, file_name, content_type, file_info, progress_listener
+        )
+        self.continue_large_file_id = continue_large_file_id
+        self.max_queue_size = max_queue_size
+
+    def execute_plan(self, emerge_plan):
+        total_length = emerge_plan.get_total_length()
+
+        if total_length is not None and self.MAX_LARGE_FILE_SIZE < total_length:
```

Comment:
```suggestion
        if total_length is not None and total_length > self.MAX_LARGE_FILE_SIZE:
```
pretty sure pep-8 says which way we should compare variables